tools/AzureRM/AzureRM.psm1

```diff
@@ -1,260 +0,0 @@
-
- .Description
-  Installs all the available Azure Resource Manager cmdlet modules that have a matching major version.
-
- .Parameter MajorVersion
-  Specifies the major version.
-
- .Parameter Repository
-  Limit the search for "AzureRM" cmdlets in a specific repository.
- 
- .Parameter Scope
-  Specifies the parameter scope.
-
- .Parameter Force
-  Force download and installation of modules already installed.
-#>
-function Update-AzureRM
-{
-  
-  param(
-  [Parameter(Position=0, Mandatory = $false)]
-  [string]
-  $MajorVersion,
-  [Parameter(Position=1, Mandatory = $false)]
-  [string]
-  $Repository = "PSGallery",
-  [Parameter(Position=2, Mandatory = $false)]
-  [ValidateSet("CurrentUser","AllUsers")]
-  [string]
-  $Scope = "AllUsers",
-  [switch]
-  $Force = $false)
-
-  Test-AdminRights $Scope
-  CheckIncompatibleVersion($Force.IsPresent)
-
-  Write-Output "Installing AzureRM modules."
-
-  $_InstallationPolicy = (Get-PSRepository -Name $Repository).InstallationPolicy
-  $script:InstallCounter = 0
-
-  try 
-  {
-    Set-PSRepository -Name $Repository -InstallationPolicy Trusted
-
-    # Update Profile and Storage
-    Install-ModuleWithVersionCheck "AzureRM.Profile" $AzureRMDependencies["AzureRM.Profile"] $Repository $Scope -Force:$force
-    Install-ModuleWithVersionCheck "Azure.Storage" $AzureRMDependencies["Azure.Storage"] $Repository $Scope -Force:$force
-
-    # Start new job
-    $AzureRMModules.Keys | ForEach {
-      $_MinVer = $MajorVersion 
-      if(!$MajorVersion) {
-        $_MinVer = $AzureRMModules[$_]
-      }
-      Install-ModuleWithVersionCheck $_ $_MinVer $Repository $Scope -Force:$force
-    }    
-  } finally {
-    # Clean up
-    Set-PSRepository -Name $Repository -InstallationPolicy $_InstallationPolicy
-  }
-}
-
-<#
- .Synopsis
-  Import Azure Resource Manager cmdlet modules
-
- .Description
-  Imports all the Azure Resource Manager cmdlet modules that have a matching major version.
-
- .Parameter MajorVersion
-  Specifies the major version.
-#>
-function Import-AzureRM
-{
-  param(
-  [Parameter(Position=0, Mandatory = $false)]
-  [string]
-  $MajorVersion )
-  Write-Output "Importing AzureRM modules."
-
-  $AzureRMModules.Keys | ForEach {
-    $moduleName = $_
-    $_MinVer = $MajorVersion 
-    if(!$MajorVersion) {
-      $_MinVer = $AzureRMModules[$_]
-    }
-    $_MaxVer = "$($_MinVer.Split(".")[0]).9999.9999.9999"
-
-    $_MatchedModule = Get-InstalledModule -Name $moduleName -MinimumVersion $_MinVer -MaximumVersion $_MaxVer -ErrorAction Ignore | where {$_.Name -eq $moduleName}
-    if ($_MatchedModule -ne $null) {
-      try {
-        Import-Module -Name $_MatchedModule.Name -RequiredVersion $_MatchedModule.Version -ErrorAction Stop
-        Write-Output "$moduleName imported..." 
-      } catch {
-        Write-Warning "Skipping $Name module..."
-        Write-Warning $_
-      }
-    }
-  }
-}
-
-function Uninstall-ModuleWithVersionCheck([string]$Name,[string]$MinVersion)
```

Comment:
We *may* decide that we need an uninstall cmdlet - currently, if you uninstall the manifest module, the dependent modules remain.  We may retain an 'Uninstall-AzureRm' script which uninstalls the module and its dependencies (not sure if it's possible for a cmdlet to remove all associated modules.  Regardless, we should decide on some acceptance criteria before merging.